src/components/AddBook.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import {
+  Form, Button, Row, Col,
+} from 'react-bootstrap';
+import { useDispatch } from 'react-redux';
+import { addBook, fetchBooks } from '../redux/books/booksSlice';
+
+const AddBook = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const handleAddBook = async () => {
+    if (!title || !author || !category) {
+      setErrorMessage('Please fill in all the fields before adding the book.');
+      return;
+    }
+
+    try {
+      await dispatch(addBook({ title, author, category }));
+      setTitle('');
+      setAuthor('');
+      setCategory('');
+      setErrorMessage('');
+      await dispatch(fetchBooks());
+    } catch (error) {
+      setErrorMessage('Error adding book. Please try again later.');
+    }
+  };
+
+  return (
+    <Form className="add-form">
+      <h3>ADD NEW BOOK</h3>
+      <Row className="align-items-center flex">
+        <Col md={4}>
+          <Form.Control
+            type="text"
+            placeholder="Book Title"
+            value={title}
+            onChange={(e) => setTitle(e.target.value)}
+          />
+        </Col>
+        <Col md={3}>
+          <Form.Control
+            type="text"
+            placeholder="Book Author"
+            value={author}
+            onChange={(e) => setAuthor(e.target.value)}
+          />
+        </Col>
+        <Col md={3}>
+          <Form.Control
+            type="text"
+            placeholder="Book Category"
+            value={category}
+            onChange={(e) => setCategory(e.target.value)}
+          />
+        </Col>
+        {errorMessage && <div className="text-danger error-message">{errorMessage}</div>}
+        <Col className="px-3 ms-4">
+          <Button className="btn-lg" variant="primary btn-lg" size="lg" block onClick={handleAddBook}>
+            Add Book
+          </Button>
```

Comment:
- [ ] As mentioned earlier by the previous reviewers, this button add should be aligned to the layout about. It is important to follow the design as much as possible 🔢 
You can fix that by using a custom margin in your css file.

Here is the design I am refering to 👇🏽 

![Screenshot from 2023-09-03 00-56-39](https://github.com/axcell-0/bookstore/assets/10439283/962edcc6-da45-4c55-9209-d9fac058147b)